art/estimators/certification/randomized_smoothing/randomized_smoothing_numpy.py

```diff
@@ -0,0 +1,148 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2018
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module implements Randomized Smoothing applied to classifier predictions.
+
+| Paper link: https://arxiv.org/abs/1902.02918
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+
+import numpy as np
+
+from art.config import ART_NUMPY_DTYPE
+from art.estimators.certification.randomized_smoothing.randomized_smoothing import RandomizedSmoothingMixin
+from art.estimators.classification import ClassifierMixin, ClassGradientsMixin
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from art.utils import CLASSIFIER_LOSS_GRADIENTS_TYPE
+
+logger = logging.getLogger(__name__)
+
+
+class RandomizedSmoothing(RandomizedSmoothingMixin, ClassGradientsMixin, ClassifierMixin):
+    """
+    Implementation of Randomized Smoothing applied to classifier predictions and gradients, as introduced
+    in Cohen et al. (2019).
+
+    | Paper link: https://arxiv.org/abs/1902.02918
+    """
+
+    def __init__(self, classifier: "CLASSIFIER_LOSS_GRADIENTS_TYPE",
+                 sample_size: int,
+                 scale: float = 0.1,
+                 alpha: float = 0.001):
+        """
+        Create a randomized smoothing wrapper.
+        :param classifier: The Classifier we want to wrap the functionality for the purpose of smoothing.
+        :param sample_size: Number of samples for smoothing
+        :type sample_size: `int`
+        :param scale: Standard deviation of Gaussian noise added.
+        :type scale: `float`
+        :param alpha: The failure probability of smoothing
+        :type alpha: `float`
+        """
+        super().__init__(classifier)
+        self.classifier = classifier
```

Comment:
```suggestion
```

Reply:
This line is required I think. The inherited classes don't have the classifier object defined. If I remove this line, I get an error `object has no attribute 'classifier'`. This isn't a problem for the framework specific classifiers as they initialize a framework specific classifier when it is required.

Reply:
Ok, now I see, in that case we can ignore this comment.